File: khalti-android/src/main/java/khalti/checkOut/api/KhaltiApi.java
package khalti.checkOut.api;


import java.util.HashMap;
import java.util.Map;

import khalti.checkOut.EBanking.helper.BaseListPojo;
import khalti.checkOut.Wallet.helper.WalletConfirmPojo;
import khalti.checkOut.Wallet.helper.WalletInitPojo;
import khalti.checkOut.helper.MerchantPreferencePojo;
import retrofit2.Response;
import retrofit2.http.FieldMap;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;
import retrofit2.http.QueryMap;
import retrofit2.http.Url;
import rx.Observable;

public interface KhaltiApi {
    @GET
    Observable<Response<BaseListPojo>> getBanks(@Url String url, @QueryMap Map<String, Object> queryMap);

    @GET
    Observable<Response<MerchantPreferencePojo>> getPreference(@Url String url, @Header("Authorization") String publicKey);

    @POST
    @FormUrlEncoded
    Observable<Response<WalletInitPojo>> initiatePayment(@Url String url, @FieldMap HashMap<String, Object> dataMap);

    @POST
    @FormUrlEncoded
    Observable<Response<WalletConfirmPojo>> confirmPayment(@Url String url, @FieldMap HashMap<String, Object> dataMap);
}

File: khalti-android/src/main/java/com/khalti/checkOut/api/KhaltiApi.kt
package com.khalti.checkOut.api


import com.khalti.checkOut.banking.helper.BaseListPojo
import com.khalti.checkOut.wallet.helper.WalletConfirmPojo
import com.khalti.checkOut.wallet.helper.WalletInitPojo
import com.khalti.checkOut.helper.MerchantPreferencePojo
import retrofit2.Response
import retrofit2.http.*

@JvmSuppressWildcards
interface KhaltiApi {

    @GET
    suspend fun getBanks(@Url url: String, @QueryMap queryMap: Map<String, Any>): Response<BaseListPojo>

    @GET
    suspend fun getPreference(@Url url: String, @Header("Authorization") publicKey: String): Response<MerchantPreferencePojo>

    @POST
    @FormUrlEncoded
    suspend fun initiatePayment(@Url url: String, @FieldMap dataMap: Map<String, Any>): Response<WalletInitPojo>

    @POST
    @FormUrlEncoded
    suspend fun confirmPayment(@Url url: String, @FieldMap dataMap: Map<String, Any>): Response<WalletConfirmPojo>
}